jrt/tests/com/yahoo/jrt/TlsDetectionTest.java

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.jrt;
+
+public class TlsDetectionTest {
+
+    static private String message(byte[] data, boolean actual) {
+        String msg = "[";
+        String delimiter = "";
+        for (byte b: data) {
+            msg += delimiter + (b & 0xff);
+            delimiter = ", ";
+        }
+        if (actual) {
+            msg += "] wrongfully detected as tls";
+        } else {
+            msg += "] wrongfully rejected as not tls";
+        }
+        return msg;
+    }
+
+    static private void checkTls(boolean expect, int ... values) {
+        byte[] data = new byte[values.length];
+        for (int i = 0; i < data.length; i++) {
+            data[i] = (byte) values[i];
+        }
+        boolean actual = MaybeTlsCryptoSocket.looksLikeTlsToMe(data);
+        if(actual != expect) {
```

Comment:
Consdier replacing with `org.junit.Assert.assertEquals()`

Reply:
that would produce a much worse error message when the test fails

Reply:
There is an overload of `assertEquals()` taking a custom error message.

Reply:
I tried, it made the error less readable

Reply:
will adjust message to make it work